lint-install.go

```diff
@@ -255,24 +255,28 @@ func goLintCmd(root string, level string, fix bool) string {
 
 	klog.Infof("found %d modules within %s: %s", len(found), root, found)
 	if len(found) == 0 || (len(found) == 1 && found[0] == strings.Trim(root, "/")) {
-		return fmt.Sprintf("out/linters/golangci-lint-$(GOLINT_VERSION)-$(LINT_ARCH) run%s", suffix)
+		return fmt.Sprintf("$(GOLANGCI_LINT_BIN) run%s", suffix)
 	}
 
-	return fmt.Sprintf(`find . -name go.mod -execdir "$(LINT_ROOT)/out/linters/golangci-lint-$(GOLINT_VERSION)-$(LINT_ARCH)" run -c "$(GOLINT_CONFIG)"%s \;`, suffix)
+	return fmt.Sprintf(`find . -name go.mod -execdir "$(GOLANGCI_LINT_BIN)" run -c "$(GOLINT_CONFIG)"%s \;`, suffix)
 }
 
 // shellLintCmd returns the appropriate shell lint command for a project.
 func shellLintCmd(_ string, level string, fix bool) string {
 	suffix := ""
 
 	if fix {
-		// patch(1) doesn't support patching from stdin on all platforms, so we use git apply instead
-		suffix = " -f diff | git apply -p2 -"
+		// Use git apply instead of patch(1) because it doesn't support patching from stdin on all platforms.
+		// Everything after the first | is so we don't call git apply if there's nothing to apply, which would cause git apply to return an error.
+		// This is the cross platform way to do what gnu xargs does with its --no-run-if-empty switch.
```

Comment:
crossplatform as in tested in linux and macos. Would definitely appreciate checking on other oses.